gradle/gradle_properties.go

```diff
@@ -0,0 +1,42 @@
+package gradle
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/buildpacks/libcnb"
+)
+
+type PropertiesFile struct {
+	Binding    libcnb.Binding
+	GradleHome string
+}
+
+func (p PropertiesFile) Contribute(layer libcnb.Layer) (libcnb.Layer, error) {
+	path, ok := p.Binding.SecretFilePath("gradle.properties")
+	if !ok {
+		return layer, nil
+	}
+
+	gradlePropertiesPath := filepath.Join(p.GradleHome, "gradle.properties")
+	if err := os.Symlink(path, gradlePropertiesPath); os.IsExist(err) {
+		err = os.Remove(gradlePropertiesPath)
```

Comment:
If I build with a binding, and then rebuild without a binding, do I have a broken symlink handing around? given this contributor only executes when a binding in present...

Reply:
Yes, there will still be a broken symlink. I did some testing and Gradle didn't seem to mind. It carried on even after I removed the binding. I can confirm because it switched the log level back to the default, not what was in my file & no error messages were printed.

I think I could perhaps delete the file, but I was looking and it could get tricky, plus Gradle doesn't seem to care so at this point I think I'd rather just leave it there. If it causes issues later, I can circle back and clean it up.